From 790cf4b5d69d1eb359e4664a18d6191bf0d4b753 Mon Sep 17 00:00:00 2001
From: Haibo Chen <haibo.chen@nxp.com>
Date: Thu, 14 Jun 2018 16:00:26 +0800
Subject: [PATCH 4014/5242] MLK-18612-2 ARM64: dts: fsl-imx8x-arm2: add ADC
 support

commit  7f085c96c0d64a3eea75aad3ebb8ede31b42bce9 from
https://source.codeaurora.org/external/imx/linux-imx.git

Add ADC support for imx8qxp b0 validation board, just enable ADC_IN0
and ADC_IN1, to support these two adc external analog input, need do
the following hardware rework: change register R335 and R338 from A
side to B side.

Signed-off-by: Haibo Chen <haibo.chen@nxp.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 arch/arm64/boot/dts/freescale/fsl-imx8x-arm2.dtsi |   21 +++++++++++++++++++++
 1 file changed, 21 insertions(+)

diff --git a/arch/arm64/boot/dts/freescale/fsl-imx8x-arm2.dtsi b/arch/arm64/boot/dts/freescale/fsl-imx8x-arm2.dtsi
index b38a2c3..8107266 100644
--- a/arch/arm64/boot/dts/freescale/fsl-imx8x-arm2.dtsi
+++ b/arch/arm64/boot/dts/freescale/fsl-imx8x-arm2.dtsi
@@ -62,6 +62,13 @@
 			gpio = <&max7322 0 GPIO_ACTIVE_HIGH>;
 			enable-active-high;
 		};
+
+		reg_adc_vref_1v8: adc_vref_1v8 {
+			compatible = "regulator-fixed";
+			regulator-name = "vref_1v8";
+			regulator-min-microvolt = <1800000>;
+			regulator-max-microvolt = <1800000>;
+		};
 	};
 
 	sound-cs42888 {
@@ -86,6 +93,13 @@
 	status = "okay";
 };
 
+&adc0 {
+	pinctrl-names = "default";
+	pinctrl-0 = <&pinctrl_adc0>;
+	vref-supply = <&reg_adc_vref_1v8>;
+	status = "okay";
+};
+
 &amix {
 	status = "okay";
 };
@@ -151,6 +165,13 @@
 			>;
 		};
 
+		pinctrl_adc0: adc0grp {
+			fsl,pins = <
+				SC_P_ADC_IN0_ADMA_ADC_IN0		0x60
+				SC_P_ADC_IN1_ADMA_ADC_IN1		0x60
+			>;
+		};
+
 		pinctrl_csi0_lpi2c0: csi0lpi2c0grp {
 			fsl,pins = <
 				SC_P_MIPI_CSI0_I2C0_SCL_MIPI_CSI0_I2C0_SCL	0xc2000020
-- 
1.7.9.5

